Source/WTF/wtf/text/StringImpl.cpp

```diff
@@ -129,26 +129,25 @@ StringImpl::~StringImpl()
             symbolRegistry->remove(*symbol.asRegisteredSymbolImpl());
     }
 
-    BufferOwnership ownership = bufferOwnership();
-
-    if (ownership == BufferInternal)
-        return;
-    if (ownership == BufferOwned) {
+    switch (bufferOwnership()) {
+    case BufferInternal:
+        break;
+    case BufferOwned:
         // We use m_data8, but since it is a union with m_data16 this works either way.
         ASSERT(m_data8);
         StringImplMalloc::free(const_cast<LChar*>(m_data8));
-        return;
-    }
-    if (ownership == BufferExternal) {
+        break;
+    case BufferExternal: {
         auto* external = static_cast<ExternalStringImpl*>(this);
         external->freeExternalBuffer(const_cast<LChar*>(m_data8), sizeInBytes());
         external->m_free.~ExternalStringImplFreeFunction();
-        return;
+        break;
```

Comment:
can all these breaks be returns instead?

Reply:
They could. Why would that be better?

Reply:
To me personally it's more readable since it implies there's no more logic after the switch-case but ya it's not objectively better 😄